### PR TITLE
Fix for TBATS issues on M3 dataset

### LINKED ISSUE #3233: [ENH]: TBATS does not work well for certain datasets
### Describe the feature you want to add to this project

When [benchmarking TBATS on M3 datasets](https://github.com/pycaret/benchmarks/blob/main/README.md), a few types of errors occur

1. Singleton array array(nan) cannot be considered a valid collection.
2. Input contains infinity or a value too large for dtype('float64').
3. y must have only positive values for box-cox transformation.

```
2023-01-07 09:49:21,958 INFO worker.py:1518 -- Started a local Ray instance.
(forecast_create_model pid=6320) Error occurred for ID: Q106 when trying main model: Singleton array array(nan) cannot be considered a valid collection.
(forecast_create_model pid=6320) Trying backup model for ID: Q106
(forecast_create_model pid=6320) Error occurred for ID: Q115 when trying main model: Input contains infinity or a value too large for dtype('float64').
(forecast_create_model pid=6320) Trying backup model for ID: Q115
(forecast_create_model pid=14196) Error occurred for ID: Q137 when trying main model: Singleton array array(nan) cannot be considered a valid collection.
(forecast_create_model pid=14196) Trying backup model for ID: Q137
(forecast_create_model pid=2084) Error occurred for ID: Q16 when trying main model: Input contains infinity or a value too large for dtype('float64').
(forecast_create_model pid=2084) Trying backup model for ID: Q16
(forecast_create_model pid=31000) Error occurred for ID: Q167 when trying main model: Singleton array array(nan) cannot be considered a valid collection.
(forecast_create_model pid=31000) Trying backup model for ID: Q167
(forecast_create_model pid=12072) Error occurred for ID: Q169 when trying main model: Singleton array array(nan) cannot be considered a valid collection.
(forecast_create_model pid=12072) Trying backup model for ID: Q169
(forecast_create_model pid=14196) Error occurred for ID: Q172 when trying main model: Input contains infinity or a value too large for dtype('float64').
(forecast_create_model pid=14196) Trying backup model for ID: Q172
(forecast_create_model pid=14196) Error occurred for ID: Q186 when trying main model: Input contains infinity or a value too large for dtype('float64').
(forecast_create_model pid=14196) Trying backup model for ID: Q186
(forecast_create_model pid=12072) Error occurred for ID: Q193 when trying main model: Input contains infinity or a value too large for dtype('float64').
(forecast_create_model pid=12072) Trying backup model for ID: Q193
(forecast_create_model pid=31000) Error occurred for ID: Q209 when trying main model: Singleton array array(nan) cannot be considered a valid collection.
(forecast_create_model pid=31000) Trying backup model for ID: Q209
(forecast_create_model pid=2084) Error occurred for ID: Q244 when trying main model: Input contains infinity or a value too large for dtype('float64').
(forecast_create_model pid=2084) Trying backup model for ID: Q244
(forecast_create_model pid=31000) Error occurred for ID: Q257 when trying main model: Input contains infinity or a value too large for dtype('float64').
(forecast_create_model pid=31000) Trying backup model for ID: Q257
(forecast_create_model pid=12072) Error occurred for ID: Q305 when trying main model: Singleton array array(nan) cannot be considered a valid collection.
(forecast_create_model pid=12072) Trying backup model for ID: Q305
(forecast_create_model pid=11040) Error occurred for ID: Q310 when trying main model: Input contains infinity or a value too large for dtype('float64').
(forecast_create_model pid=11040) Trying backup model for ID: Q310
(forecast_create_model pid=11040) Error occurred for ID: Q333 when trying main model: Singleton array array(nan) cannot be considered a valid collection.
(forecast_create_model pid=11040) Trying backup model for ID: Q333
(forecast_create_model pid=11232) Error occurred for ID: Q343 when trying main model: Input contains infinity or a value too large for dtype('float64').
(forecast_create_model pid=11232) Trying backup model for ID: Q343
(forecast_create_model pid=12072) Error occurred for ID: Q344 when trying main model: Input contains infinity or a value too large for dtype('float64').
(forecast_create_model pid=12072) Trying backup model for ID: Q344
(forecast_create_model pid=11040) Error occurred for ID: Q341 when trying main model: Input contains infinity or a value too large for dtype('float64').
(forecast_create_model pid=11040) Trying backup model for ID: Q341
(forecast_create_model pid=31000) Error occurred for ID: Q362 when trying main model: Input contains infinity or a value too large for dtype('float64').
(forecast_create_model pid=31000) Trying backup model for ID: Q362
(forecast_create_model pid=12072) Error occurred for ID: Q363 when trying main model: Singleton array array(nan) cannot be considered a valid collection.
(forecast_create_model pid=12072) Trying backup model for ID: Q363
(forecast_create_model pid=2084) Error occurred for ID: Q371 when trying main model: Input contains infinity or a value too large for dtype('float64').
(forecast_create_model pid=2084) Trying backup model for ID: Q371
(forecast_create_model pid=18780) Error occurred for ID: Q379 when trying main model: Singleton array array(nan) cannot be considered a valid collection.
(forecast_create_model pid=18780) Trying backup model for ID: Q379
(forecast_create_model pid=12072) Error occurred for ID: Q385 when trying main model: Singleton array array(nan) cannot be considered a valid collection.
(forecast_create_model pid=12072) Trying backup model for ID: Q385
(forecast_create_model pid=11040) Error occurred for ID: Q387 when trying main model: Singleton array array(nan) cannot be considered a valid collection.
(forecast_create_model pid=11040) Trying backup model for ID: Q387
(forecast_create_model pid=2084) Error occurred for ID: Q382 when trying main model: y must have only positive values for box-cox transformation.
(forecast_create_model pid=2084) Trying backup model for ID: Q382
(forecast_create_model pid=6320) Error occurred for ID: Q393 when trying main model: y must have only positive values for box-cox transformation.
(forecast_create_model pid=6320) Trying backup model for ID: Q393
(forecast_create_model pid=18780) Error occurred for ID: Q394 when trying main model: y must have only positive values for box-cox transformation.
(forecast_create_model pid=18780) Trying backup model for ID: Q394
(forecast_create_model pid=2084) Error occurred for ID: Q396 when trying main model: y must have only positive values for box-cox transformation.
(forecast_create_model pid=2084) Trying backup model for ID: Q396
(forecast_create_model pid=6320) Error occurred for ID: Q398 when trying main model: y must have only positive values for box-cox transformation.
(forecast_create_model pid=6320) Trying backup model for ID: Q398
(forecast_create_model pid=12072) Error occurred for ID: Q399 when trying main model: y must have only positive values for box-cox transformation.
(forecast_create_model pid=12072) Trying backup model for ID: Q399
(forecast_create_model pid=12072) Error occurred for ID: Q406 when trying main model: Singleton array array(nan) cannot be considered a valid collection.
(forecast_create_model pid=12072) Trying backup model for ID: Q406
(forecast_create_model pid=18780) Error occurred for ID: Q407 when trying main model: Singleton array array(nan) cannot be considered a valid collection.
(forecast_create_model pid=18780) Trying backup model for ID: Q407
(forecast_create_model pid=12072) Error occurred for ID: Q408 when trying main model: Singleton array array(nan) cannot be considered a valid collection.
(forecast_create_model pid=12072) Trying backup model for ID: Q408
(forecast_create_model pid=31000) Error occurred for ID: Q412 when trying main model: Singleton array array(nan) cannot be considered a valid collection.
(forecast_create_model pid=31000) Trying backup model for ID: Q412
(forecast_create_model pid=12072) Error occurred for ID: Q414 when trying main model: Singleton array array(nan) cannot be considered a valid collection.
(forecast_create_model pid=12072) Trying backup model for ID: Q414
(forecast_create_model pid=18780) Error occurred for ID: Q415 when trying main model: Singleton array array(nan) cannot be considered a valid collection.
(forecast_create_model pid=18780) Trying backup model for ID: Q415
(forecast_create_model pid=14196) Error occurred for ID: Q420 when trying main model: Singleton array array(nan) cannot be considered a valid collection.
(forecast_create_model pid=14196) Trying backup model for ID: Q420
(forecast_create_model pid=11232) Error occurred for ID: Q42 when trying main model: Singleton array array(nan) cannot be considered a valid collection.
(forecast_create_model pid=11232) Trying backup model for ID: Q42
(forecast_create_model pid=11040) Error occurred for ID: Q427 when trying main model: Singleton array array(nan) cannot be considered a valid collection.
(forecast_create_model pid=11040) Trying backup model for ID: Q427
(forecast_create_model pid=18780) Error occurred for ID: Q425 when trying main model: y must have only positive values for box-cox transformation.
(forecast_create_model pid=18780) Trying backup model for ID: Q425
(forecast_create_model pid=18780) Error occurred for ID: Q43 when trying main model: Input contains infinity or a value too large for dtype('float64').
(forecast_create_model pid=18780) Trying backup model for ID: Q43
(forecast_create_model pid=6320) Error occurred for ID: Q448 when trying main model: Singleton array array(nan) cannot be considered a valid collection.
(forecast_create_model pid=6320) Trying backup model for ID: Q448
(forecast_create_model pid=12072) Error occurred for ID: Q450 when trying main model: Input contains infinity or a value too large for dtype('float64').
(forecast_create_model pid=12072) Trying backup model for ID: Q450
(forecast_create_model pid=14196) Error occurred for ID: Q463 when trying main model: y must have only positive values for box-cox transformation.
(forecast_create_model pid=14196) Trying backup model for ID: Q463
(forecast_create_model pid=11040) Error occurred for ID: Q467 when trying main model: Singleton array array(nan) cannot be considered a valid collection.
(forecast_create_model pid=11040) Trying backup model for ID: Q467
(forecast_create_model pid=11040) Error occurred for ID: Q478 when trying main model: Singleton array array(nan) cannot be considered a valid collection.
(forecast_create_model pid=11040) Trying backup model for ID: Q478
(forecast_create_model pid=12072) Error occurred for ID: Q484 when trying main model: y must have only positive values for box-cox transformation.
(forecast_create_model pid=12072) Trying backup model for ID: Q484
(forecast_create_model pid=31000) Error occurred for ID: Q483 when trying main model: y must have only positive values for box-cox transformation.
(forecast_create_model pid=31000) Trying backup model for ID: Q483
(forecast_create_model pid=11040) Error occurred for ID: Q488 when trying main model: y must have only positive values for box-cox transformation.
(forecast_create_model pid=11040) Trying backup model for ID: Q488
(forecast_create_model pid=18780) Error occurred for ID: Q496 when trying main model: Singleton array array(nan) cannot be considered a valid collection.
(forecast_create_model pid=18780) Trying backup model for ID: Q496
(forecast_create_model pid=2084) Error occurred for ID: Q501 when trying main model: Singleton array array(nan) cannot be considered a valid collection.
(forecast_create_model pid=2084) Trying backup model for ID: Q501
(forecast_create_model pid=31000) Error occurred for ID: Q502 when trying main model: Singleton array array(nan) cannot be considered a valid collection.
(forecast_create_model pid=31000) Trying backup model for ID: Q502
(forecast_create_model pid=11232) Error occurred for ID: Q503 when trying main model: Singleton array array(nan) cannot be considered a valid collection.
(forecast_create_model pid=11232) Trying backup model for ID: Q503
(forecast_create_model pid=18780) Error occurred for ID: Q504 when trying main model: Input contains infinity or a value too large for dtype('float64').
(forecast_create_model pid=18780) Trying backup model for ID: Q504
(forecast_create_model pid=6320) Error occurred for ID: Q498 when trying main model: y must have only positive values for box-cox transformation.
(forecast_create_model pid=6320) Trying backup model for ID: Q498
(forecast_create_model pid=18780) Error occurred for ID: Q508 when trying main model: Input contains infinity or a value too large for dtype('float64').
(forecast_create_model pid=18780) Trying backup model for ID: Q508
(forecast_create_model pid=18780) Error occurred for ID: Q512 when trying main model: Input contains infinity or a value too large for dtype('float64').
(forecast_create_model pid=18780) Trying backup model for ID: Q512
(forecast_create_model pid=12072) Error occurred for ID: Q510 when trying main model: Input contains infinity or a value too large for dtype('float64').
(forecast_create_model pid=12072) Trying backup model for ID: Q510
(forecast_create_model pid=6320) Error occurred for ID: Q51 when trying main model: y must have only positive values for box-cox transformation.
(forecast_create_model pid=6320) Trying backup model for ID: Q51
(forecast_create_model pid=18780) Error occurred for ID: Q513 when trying main model: Input contains infinity or a value too large for dtype('float64').
(forecast_create_model pid=18780) Trying backup model for ID: Q513
(forecast_create_model pid=31000) Error occurred for ID: Q52 when trying main model: Singleton array array(nan) cannot be considered a valid collection.
(forecast_create_model pid=31000) Trying backup model for ID: Q52
(forecast_create_model pid=12072) Error occurred for ID: Q526 when trying main model: Input contains infinity or a value too large for dtype('float64').
(forecast_create_model pid=12072) Trying backup model for ID: Q526
(forecast_create_model pid=12072) Error occurred for ID: Q541 when trying main model: Input contains infinity or a value too large for dtype('float64').
(forecast_create_model pid=12072) Trying backup model for ID: Q541
(forecast_create_model pid=12072) Error occurred for ID: Q542 when trying main model: Input contains infinity or a value too large for dtype('float64').
(forecast_create_model pid=12072) Trying backup model for ID: Q542
(forecast_create_model pid=14196) Error occurred for ID: Q543 when trying main model: Input contains infinity or a value too large for dtype('float64').
(forecast_create_model pid=14196) Trying backup model for ID: Q543
(forecast_create_model pid=6320) Error occurred for ID: Q548 when trying main model: Input contains infinity or a value too large for dtype('float64').
(forecast_create_model pid=6320) Trying backup model for ID: Q548
(forecast_create_model pid=6320) Error occurred for ID: Q551 when trying main model: Input contains infinity or a value too large for dtype('float64').
(forecast_create_model pid=6320) Trying backup model for ID: Q551
(forecast_create_model pid=18780) Error occurred for ID: Q549 when trying main model: Input contains infinity or a value too large for dtype('float64').
(forecast_create_model pid=18780) Trying backup model for ID: Q549
(forecast_create_model pid=14196) Error occurred for ID: Q556 when trying main model: Input contains infinity or a value too large for dtype('float64').
(forecast_create_model pid=14196) Trying backup model for ID: Q556
(forecast_create_model pid=18780) Error occurred for ID: Q576 when trying main model: Input contains infinity or a value too large for dtype('float64').
(forecast_create_model pid=18780) Trying backup model for ID: Q576
(forecast_create_model pid=14196) Error occurred for ID: Q577 when trying main model: Input contains infinity or a value too large for dtype('float64').
(forecast_create_model pid=14196) Trying backup model for ID: Q577
(forecast_create_model pid=18780) Error occurred for ID: Q579 when trying main model: Input contains infinity or a value too large for dtype('float64').
(forecast_create_model pid=18780) Trying backup model for ID: Q579
(forecast_create_model pid=11232) Error occurred for ID: Q581 when trying main model: Input contains infinity or a value too large for dtype('float64').
(forecast_create_model pid=11232) Trying backup model for ID: Q581
(forecast_create_model pid=12072) Error occurred for ID: Q582 when trying main model: Input contains infinity or a value too large for dtype('float64').
(forecast_create_model pid=12072) Trying backup model for ID: Q582
(forecast_create_model pid=31000) Error occurred for ID: Q584 when trying main model: Input contains infinity or a value too large for dtype('float64').
(forecast_create_model pid=31000) Trying backup model for ID: Q584
(forecast_create_model pid=11232) Error occurred for ID: Q586 when trying main model: Input contains infinity or a value too large for dtype('float64').
(forecast_create_model pid=11232) Trying backup model for ID: Q586
(forecast_create_model pid=11232) Error occurred for ID: Q591 when trying main model: Input contains infinity or a value too large for dtype('float64').
(forecast_create_model pid=11232) Trying backup model for ID: Q591
(forecast_create_model pid=2084) Error occurred for ID: Q6 when trying main model: Input contains infinity or a value too large for dtype('float64').
(forecast_create_model pid=2084) Trying backup model for ID: Q6
(forecast_create_model pid=2084) Error occurred for ID: Q603 when trying main model: Singleton array array(nan) cannot be considered a valid collection.
(forecast_create_model pid=2084) Trying backup model for ID: Q603
(forecast_create_model pid=12072) Error occurred for ID: Q610 when trying main model: y must have only positive values for box-cox transformation.
(forecast_create_model pid=12072) Trying backup model for ID: Q610
(forecast_create_model pid=11040) Error occurred for ID: Q620 when trying main model: y must have only positive values for box-cox transformation.
(forecast_create_model pid=11040) Trying backup model for ID: Q620
(forecast_create_model pid=31000) Error occurred for ID: Q623 when trying main model: Input contains infinity or a value too large for dtype('float64').
(forecast_create_model pid=31000) Trying backup model for ID: Q623
(forecast_create_model pid=11040) Error occurred for ID: Q626 when trying main model: Input contains infinity or a value too large for dtype('float64').
(forecast_create_model pid=11040) Trying backup model for ID: Q626
(forecast_create_model pid=2084) Error occurred for ID: Q647 when trying main model: Singleton array array(nan) cannot be considered a valid collection.
(forecast_create_model pid=2084) Trying backup model for ID: Q647
(forecast_create_model pid=11232) Error occurred for ID: Q652 when trying main model: Input contains infinity or a value too large for dtype('float64').
(forecast_create_model pid=11232) Trying backup model for ID: Q652
(forecast_create_model pid=18780) Error occurred for ID: Q657 when trying main model: Input contains infinity or a value too large for dtype('float64').
(forecast_create_model pid=18780) Trying backup model for ID: Q657
(forecast_create_model pid=11040) Error occurred for ID: Q661 when trying main model: Singleton array array(nan) cannot be considered a valid collection.
(forecast_create_model pid=11040) Trying backup model for ID: Q661
(forecast_create_model pid=31000) Error occurred for ID: Q669 when trying main model: Input contains infinity or a value too large for dtype('float64').
(forecast_create_model pid=31000) Trying backup model for ID: Q669
(forecast_create_model pid=14196) Error occurred for ID: Q668 when trying main model: Input contains infinity or a value too large for dtype('float64').
(forecast_create_model pid=14196) Trying backup model for ID: Q668
(forecast_create_model pid=11040) Error occurred for ID: Q674 when trying main model: Input contains infinity or a value too large for dtype('float64').
(forecast_create_model pid=11040) Trying backup model for ID: Q674
(forecast_create_model pid=14196) Error occurred for ID: Q678 when trying main model: Input contains infinity or a value too large for dtype('float64').
(forecast_create_model pid=14196) Trying backup model for ID: Q678
(forecast_create_model pid=18780) Error occurred for ID: Q685 when trying main model: y must have only positive values for box-cox transformation.
(forecast_create_model pid=18780) Trying backup model for ID: Q685
(forecast_create_model pid=2084) Error occurred for ID: Q693 when trying main model: Input contains infinity or a value too large for dtype('float64').
(forecast_create_model pid=2084) Trying backup model for ID: Q693
(forecast_create_model pid=12072) Error occurred for ID: Q690 when trying main model: y must have only positive values for box-cox transformation.
(forecast_create_model pid=12072) Trying backup model for ID: Q690
(forecast_create_model pid=11232) Error occurred for ID: Q695 when trying main model: y must have only positive values for box-cox transformation.
(forecast_create_model pid=11232) Trying backup model for ID: Q695
(forecast_create_model pid=31000) Error occurred for ID: Q701 when trying main model: y must have only positive values for box-cox transformation.
(forecast_create_model pid=31000) Trying backup model for ID: Q701
(forecast_create_model pid=12072) Error occurred for ID: Q703 when trying main model: Input contains infinity or a value too large for dtype('float64').
(forecast_create_model pid=12072) Trying backup model for ID: Q703
(forecast_create_model pid=11232) Error occurred for ID: Q706 when trying main model: Input contains infinity or a value too large for dtype('float64').
(forecast_create_model pid=11232) Trying backup model for ID: Q706
(forecast_create_model pid=11232) Error occurred for ID: Q717 when trying main model: Input contains infinity or a value too large for dtype('float64').
(forecast_create_model pid=11232) Trying backup model for ID: Q717
(forecast_create_model pid=12072) Error occurred for ID: Q724 when trying main model: Input contains infinity or a value too large for dtype('float64').
(forecast_create_model pid=12072) Trying backup model for ID: Q724
(forecast_create_model pid=11040) Error occurred for ID: Q722 when trying main model: Input contains infinity or a value too large for dtype('float64').
(forecast_create_model pid=11040) Trying backup model for ID: Q722
(forecast_create_model pid=11040) Error occurred for ID: Q727 when trying main model: y must have only positive values for box-cox transformation.
(forecast_create_model pid=11040) Trying backup model for ID: Q727
(forecast_create_model pid=14196) Error occurred for ID: Q736 when trying main model: y must have only positive values for box-cox transformation.
(forecast_create_model pid=14196) Trying backup model for ID: Q736
(forecast_create_model pid=12072) Error occurred for ID: Q94 when trying main model: y must have only positive values for box-cox transformation.
(forecast_create_model pid=12072) Trying backup model for ID: Q94
```

### Describe your proposed solution

Check for the root cause of these errors (could be negative values in the dataset) and take appropriate action for TBATS. It may be possible that the model has to be disabled or the hyperparameters have to be set appropriately depending on the data properties.

### Describe alternatives you've considered, if relevant

_No response_

### Additional context

_No response_

### LINKED ISSUE #3483: [ENH]: Update TBATS version
### Describe the feature you want to add to this project

https://github.com/pycaret/pycaret/issues/3233

### Describe your proposed solution

Update TBATS version to 1.1.3 which should fix the issues noted in https://github.com/pycaret/pycaret/issues/3233
After this, the changes done in https://github.com/pycaret/pycaret/pull/3478 can be reversed/cleaned up to allow box-cox transformation in create_model also. 

### Describe alternatives you've considered, if relevant

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by ngupta23**
# Related Issue or bug

- Issue is with TBATS package and has been reported here: https://github.com/intive-DataScience/tbats/issues/40
- Closes #3233
- Closes #3483 

# Describe the changes you've made

- Updated version of TBATS that fixes this - based on https://github.com/intive-DataScience/tbats/issues/40

# Type of change

Please delete options that are not relevant.
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Locally tested, GitHub regression tests should pass

# Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

File: pycaret/containers/models/time_series.py [hunk 1]
```diff
@@ -1152,32 +1153,36 @@ def _set_args(self) -> dict:
 
     @property
     def _set_tune_grid(self) -> dict:
+        # None considers both True and False and selects the best one.
         tune_grid = {
-            "use_damped_trend": [True, False],
-            "use_trend": [True, False],
             "sp": [self.sp],
+            "use_box_cox": [None],
```
Comment (Yard1) [inline pycaret/containers/models/time_series.py:1159]: should this be included here?

Comment (ngupta23) [reply pycaret/containers/models/time_series.py:1159]: `None` tries out both True and False per their documentation and picks the best one.

Comment (Yard1) [reply pycaret/containers/models/time_series.py:1159]: But if it's `True` it will error out, will it not?

Comment (ngupta23) [reply pycaret/containers/models/time_series.py:1159]: Yes, but not in all cases. I wanted to keep it in tune model for cases where it works, but let it error not for other cases, but just not in `create_model`. Without the True option in at least one part of the flow, the "B" part of TBATS and BATS is not functional so I don't want to disable it completely.

**Pull request merged**